core/src/main/java/org/kiwix/kiwixmobile/core/bookmark/BookmarksActivity.kt

```diff
@@ -0,0 +1,273 @@
+        mode: ActionMode,
+        menu: Menu
+      ): Boolean = false
+
+      override fun onActionItemClicked(
+        mode: ActionMode,
+        item: MenuItem
+      ): Boolean {
+        refreshAdapter = false
+        if (item.itemId == R.id.menu_context_delete) {
+          dialogShower.show(KiwixDialog.DeleteBookmarks, {
+            allBookmarks.removeAll(deleteList)
+            for (bookmark in deleteList) {
+              val position = bookmarksList.indexOf(bookmark)
+              bookmarksList.remove(bookmark)
+              bookmarksAdapter!!.notifyItemRemoved(position)
+              bookmarksAdapter!!.notifyItemRangeChanged(position, bookmarksAdapter!!.itemCount)
+            }
+            presenter.deleteBookmarks(ArrayList(deleteList))
+            mode.finish()
+          })
+          return true
+        }
+        return false
+      }
+
+      override fun onDestroyActionMode(mode: ActionMode) {
+        if (deleteList.size != 0) {
+          deleteList.clear()
+        }
+        actionMode = null
+        if (refreshAdapter) {
+          bookmarksAdapter!!.notifyDataSetChanged()
+        }
+        bookmarks_switch!!.isEnabled = true
+      }
+    }
+
+  override fun injection(coreComponent: CoreComponent) {
+    activityComponent.inject(this)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    presenter.attachView(this)
+    setContentView(R.layout.activity_bookmarks)
+    setSupportActionBar(toolbar)
+    val actionBar = supportActionBar
+    if (actionBar != null) {
+      actionBar.setDisplayHomeAsUpEnabled(true)
+      actionBar.setTitle(R.string.bookmarks)
+    }
+    setupBookmarksAdapter()
+    recycler_view!!.adapter = bookmarksAdapter
+    bookmarks_switch!!.setOnCheckedChangeListener { _: CompoundButton?, isChecked: Boolean ->
+      sharedPreferenceUtil.showBookmarksCurrentBook = !isChecked
+      presenter.loadBookmarks(sharedPreferenceUtil.showBookmarksCurrentBook)
+    }
+    bookmarks_switch!!.isChecked = !sharedPreferenceUtil.showBookmarksCurrentBook
+  }
+
+  private fun setupBookmarksAdapter() {
+    bookmarksAdapter = BookmarksAdapter(bookmarksList, deleteList, this)
+    bookmarksAdapter!!.registerAdapterDataObserver(object : AdapterDataObserver() {
+      override fun onChanged() {
+        super.onChanged()
+        no_bookmarks!!.visibility = if (bookmarksList.size == 0) View.VISIBLE else View.GONE
+      }
+    })
+  }
+
+  override fun onResume() {
+    super.onResume()
+    presenter.loadBookmarks(sharedPreferenceUtil.showBookmarksCurrentBook)
+  }
+
+  override fun onCreateOptionsMenu(menu: Menu): Boolean {
+    menuInflater.inflate(R.menu.menu_bookmarks, menu)
+    val search = menu.findItem(R.id.menu_bookmarks_search)
+      .actionView as SearchView
+    search.queryHint = getString(R.string.search_bookmarks)
+    search.setOnQueryTextListener(object :
+      SearchView.OnQueryTextListener {
```

Comment:
This can be replaced with SimpleTextListener

Reply:
I am not getting any option of SimpleTextListener.

Reply:
Look at its other usages, there are examples in the codebase